guides/Getting Started.md

```diff
@@ -0,0 +1,696 @@
+
+```elixir
+config :friends, ecto_repos: [Friends.Repo]
+```
+
+This tells our application about the repo, which will allow us to run commands such as `mix ecto.create` very soon.
+
+We've now configured our application so that it's able to make queries to our database. Let's now create our database, add a table to it, and then perform some queries.
+
+## Setting up the database
+
+To be able to query a database, it first needs to exist. We can create the database with this command:
+
+```
+mix ecto.create
+```
+
+If the database has been created successfully, then you will see this message:
+
+```
+The database for Friends.Repo has been created.
+```
+
+**NOTE:** If you get an error, you should try changing your configuration in `config/config.exs`, as it may be an authentication error.
+
+A database by itself isn't very queryable, so we will need to create a table within that database. To do that, we'll use what's referred to as a _migration_. If you've come from Active Record (or similar), you will have seen these before. A migration is a single step in the process of constructing your database.
+
+Let's create a migration now with this command:
+
+```
+mix ecto.gen.migration create_people
+```
+
+This command will generate a brand new migration file in `priv/repo/migrations`, which is empty by default:
+
+```elixir
+defmodule Friends.Repo.Migrations.CreatePeople do
+  use Ecto.Migration
+
+  def change do
+
+  end
+end
+```
+
+Let's add some code to this migration to create a new table called "people", with a few columns in it:
+
+```elixir
+defmodule Friends.Repo.Migrations.CreatePeople do
+  use Ecto.Migration
+
+  def change do
+    create table(:people) do
+      add :first_name, :string
+      add :last_name, :string
+      add :age, :integer
+    end
+  end
+end
+```
+
+This new code will tell Ecto to create a new table called people, and add three new fields: `first_name`, `last_name` and `age` to that table. The types of these fields are `string` and `integer`. (The different types that Ecto supports are covered in the [Ecto.Schema](https://hexdocs.pm/ecto/Ecto.Schema.html) documentation.)
+
+**NOTE: The naming convention for tables in Ecto databases is to use a pluralized name.**
+
+To run this migration and create the `people` table in our database, we will run this command:
+
+```
+mix ecto.migrate
+```
+
+If we found out that we made a mistake in this migration, we could run `mix ecto.rollback` to undo the changes in the migration. We could then fix the changes in the migration and run `mix ecto.migrate` again. If we ran `mix ecto.rollback` now, it would delete the table that we just created.
+
+We now have a table created in our database. The next step that we'll need to do is to create the schema.
+
+## Creating the schema
+
+The schema is an Elixir representation of data from our database. Each schema doesn't necessarily need to be tied to a table; they can be tied to database views too.
```

Comment:
Could rephrase here, perhaps: "Schemas are commonly associated with a database table, however they can be associated with a database view as well."